File: Examples/DemoObjc/HelpcrunchSDKDemoObjc/PushDelegate.m
#import "PushDelegate.h"
#import <HelpCrunchSDK/HelpCrunch.h>

@implementation PushDelegate

- (void)userNotificationCenter:(UNUserNotificationCenter *)center
       willPresentNotification:(UNNotification *)notification
         withCompletionHandler:(void (^)(UNNotificationPresentationOptions))completionHandler {

    if (![HelpCrunch userNotificationCenterShouldPresentNotification:notification]) {
        completionHandler(UNNotificationPresentationOptionNone);
    } else {
        if (@available(iOS 14.0, *)) {
            completionHandler(UNNotificationPresentationOptionList | UNNotificationPresentationOptionBanner | UNNotificationPresentationOptionSound);
        } else {
            completionHandler(UNNotificationPresentationOptionAlert);
        };
    }
}

- (void)userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNotificationResponse:(UNNotificationResponse *)response withCompletionHandler:(void (^)(void))completionHandler {
    [HelpCrunch didReceiveRemoteNotification:response.notification.request.content.userInfo];

    completionHandler();
}

@end

File: Examples/DemoSwift/HelpcrunchSDKDemoSwift/PushDelegate.swift
import Foundation
import UserNotifications
import HelpCrunchSDK

class PushDelegate {
        
    func userNotificationCenter(_ center: UNUserNotificationCenter, willPresent notification: UNNotification, withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
            
        if HelpCrunch.userNotificationCenterShouldPresent(notification) {
            if #available(iOS 14.0, *) {
                completionHandler([.list, .badge, .sound])
            } else {
                completionHandler([.alert])
            }
        } else {
            completionHandler([])
        }
    }
    
    func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
        HelpCrunch.didReceiveRemoteNotification(response.notification.request.content.userInfo)

        completionHandler()
    }
}
